src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -1029,6 +1029,31 @@ Copyright (c) .NET Foundation. All rights reserved.
     </GenerateSupportedTargetFrameworkAlias>
   </Target>
 
+  <!--
+  ============================================================
+  ValidateExecutableReferences
+  ============================================================
+  -->
+
+  <ItemGroup>
+    <AdditionalTargetFrameworkInfoProperty Include="SelfContained"/>
+    <AdditionalTargetFrameworkInfoProperty Include="_IsExecutable"/>
+  </ItemGroup>
+
+  <UsingTask TaskName="ValidateExecutableReferences" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+
+  <Target Name="ValidateExecutableReferences"
+          AfterTargets="_GetProjectReferenceTargetFrameworkProperties"
```

Comment:
Cannot make this a dependOn?

Reply:
No, `DependsOnTargets` and `AfterTargets` mean different things: https://github.com/MicrosoftDocs/visualstudio-docs/issues/2994#issuecomment-480017831